frontend_lib/src/component/ShareDownload/NewShareDownload.jsx

```diff
@@ -0,0 +1,153 @@
+import React from 'react'
+import { withTranslation } from 'react-i18next'
+import Radium from 'radium'
+import { Popover, PopoverBody } from 'reactstrap'
+import { generateRandomPassword } from '../../helper.js'
+
+const color = require('color')
+
+class NewShareDownload extends React.Component {
+  constructor (props) {
+    super(props)
+
+    this.state = {
+      passwordActive: false,
+      popoverOpen: false,
+      hidePassword: true
+    }
+  }
+
+  popoverToggle = () => {
+    this.setState({
+      popoverOpen: !this.state.popoverOpen
+    })
+  }
+
+  handleSeePassword = () => {
+    this.setState({hidePassword: !this.state.hidePassword})
+  }
+
+  handleRandomPassword = () => {
+    this.props.onChangePassword({target: {value: generateRandomPassword()}})
+    const passwordInput = document.getElementsByClassName('shareDownload__password__input')[0]
+    if (passwordInput.type === 'password') {
+      this.handleSeePassword()
+    }
+  }
+
+  handleSeePasswordInput = () => {
+    this.setState({passwordActive: true})
+  }
+
+  render () {
+    const { props, state } = this
+
+    return (
+      <div className='shareDownload__new'>
+        <div className='shareDownload__title'>
+          {props.t('New share')}
+        </div>
+
+        <div className='shareDownload__email'>
+          <textarea
+            type='text'
+            className='shareDownload__email__input form-control'
+            placeholder={props.t('Enter the email address of the recipient(s)')}
+            rows='10'
+            value={props.shareEmails}
+            onChange={props.onChangeEmails}
+            onKeyDown={props.onKeyDownEnter}
+          />
+          <button
+            type='button'
+            className='shareDownload__email__icon'
+            id='popoverMultipleEmails'
+            key='share_emails'
+            style={{':hover': {color: props.hexcolor}}}
+          >
+            <i className='fa fa-fw fa-question-circle' />
+          </button>
+          <Popover placement='bottom' isOpen={state.popoverOpen} target='popoverMultipleEmails' toggle={this.popoverToggle}>
+            <PopoverBody>{props.t('To add multiple recipients, separate the email addresses with a comma or space.')}</PopoverBody>
+          </Popover>
+        </div>
+
+        {state.passwordActive
+        ? <div className='shareDownload__password'>
```

Comment:
this is not quite readable.
Use parenthesis to always have your opening tags at the same indent level of the closing one